packages/labs/eleventy-plugin-lit/README.md

```diff
@@ -1,68 +1,396 @@
+
+```sh
 NODE_OPTIONS=--experimental-vm-modules eleventy
 ```
 
-## Client-side hydration
+### Watch mode
 
-For hydrating interactive components on the client, you should include
-`lit/experimental-hydrate-support.js` along with the component definitions used
-(it may be included in the same bundle passed to the eleventy plugin if serving
-bundled sources to the client) _at the bottom of the page_. When server-rendering components, you should load definitions after components have been parsed.
+Use [`addWatchTarget`](https://www.11ty.dev/docs/watch-serve/) to tell Eleventy
+to watch for changes in your JavaScript directory:
 
-Most browsers (excluding Chrome 90+) will also need to load and invoke the `@webcomponents/template-shadowroot` ponyfill prior to registering component definitions:
+```js
+eleventyConfig.addWatchTarget('js/');
+```
 
-```html
-<script type="module">
-  // Hydrate template-shadowroots eagerly after rendering (for browsers without
-  // native declarative shadow roots)
-  import {
-    hasNativeDeclarativeShadowRoots,
-    hydrateShadowRoots,
-  } from './node_modules/@webcomponents/template-shadowroot/template-shadowroot.js';
-  if (!hasNativeDeclarativeShadowRoots) {
-    hydrateShadowRoots(document.body);
+## Usage
+
+Whenever you use a custom element in your Eleventy Markdown and HTML files,
+`@lit-labs/eleventy-plugin-lit` will automatically render its template and
+styles directly into your HTML.
+
+For example, given a markdown file `hello.md`:
+
+```md
+# Greetings
+
+<demo-greeter name="World"></demo-greeter>
+```
+
+And a component definition `js/demo-greeter.js`:
+
+<!-- prettier-ignore-start -->
+```js
+import {LitElement, html, css} from 'lit';
+
+class DemoGreeter extends LitElement {
+  static styles = css`
+    b { color: red; }
+  `;
+
+  static properties = {
+    name: {},
+  };
+
+  render() {
+    return html`Hello <b>${this.name}</b>!`;
   }
-  // ...
-  // Load and hydrate components lazily
-  import('./_js/components.bundle.js');
-</script>
+}
+customElements.define('demo-greeter', DemoGreeter);
+```
+<!-- prettier-ignore-end -->
+
+Then the Eleventy will produce `greeting/index.html`:
+
+<!-- prettier-ignore-start -->
+```html
+<h1>Greetings</h1>
+
+<demo-greeter name="World">
+  <template shadowroot="open">
+    <style>
+      b { color: red; }
+    </style>
+  </template>
+  Hello <b>World</b>!
+</demo-greeter>
+```
+<!-- prettier-ignore-end -->
+
+### Component compatibility
+
+> 🚧 Note: Expanding this section with full details on component compatibility
+> is on the [roadmap](#roadmap). Follow
+> [#2494](https://github.com/lit/lit/issues/2494) for progress and discussion.
+> 🚧
+
+There are currently a number of restrictions that determine whether a component
+will be compatible with Lit pre-rendering, because not all of the component
+lifecycle methods are currently invoked, and the DOM APIs that can be used in
+certain lifecycle methods are restricted.
```

Comment:
I'd put the note after so it fits in the flow, or repeat it a bit: "The Lit team is working on finalizing and documenting the SSR lifecycle and restrictions."

Reply:
Done.